api/all/src/main/java/io/opentelemetry/api/GlobalOpenTelemetry.java

```diff
@@ -51,9 +51,11 @@ private GlobalOpenTelemetry() {}
    *     interface FQCN but the specified provider cannot be found.
    */
   public static OpenTelemetry get() {
-    if (globalOpenTelemetry == null) {
+    OpenTelemetry openTelemetry = globalOpenTelemetry;
+    if (openTelemetry == null) {
       synchronized (mutex) {
-        if (globalOpenTelemetry == null) {
+        openTelemetry = globalOpenTelemetry;
+        if (openTelemetry == null) {
 
           OpenTelemetry autoConfigured = maybeAutoConfigure();
```

Comment:
not related to your change (probably in a separate PR), if `maybeAutoConfigure` can be renamed would be good to have it as `makeAutoConfigureAndSet` because it is confusing that you don't set on that path.